test/eventmesh.LockManager.spec.js

```diff
@@ -0,0 +1,198 @@
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+            sinon.assert.calledOnce(releaseStub);
+          });
+      });
+      it('should succeed if no ongoing lock is there.', () => {
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        jsonStub.onCall(0).returns(noLockResp);
+        return manager.lock('fakeResource', CONST.LOCK_TYPE.WRITE)
+          .then(() => {
+            /* jshint expr: true */
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(putStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(valueStub.getCall(0).calledWithExactly(JSON.stringify(writeLockResp))).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+            sinon.assert.calledOnce(releaseStub);
+          });
+      });
+      it('should fail if an ongoing lock is there.', () => {
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        jsonStub.onCall(0).returns(writeLockResp);
+        return manager.lock('fakeResource', CONST.LOCK_TYPE.WRITE)
+          .catch(e => {
+            /* jshint expr: true */
+            expect(e.message).to.eql('Could not acquire lock for fakeResource as it is already locked.');
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+            sinon.assert.calledOnce(releaseStub);
+          });
+      });
+
+      it('loc should not fail even if release of resource lock fails.', () => {
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
+        jsonStub.onCall(0).returns(noLockResp);
+        releaseStub.onCall(0).throws(new Error('Failed for acquire lock'));
+        return manager.lock('fakeResource', CONST.LOCK_TYPE.WRITE)
+          .then(() => {
+            /* jshint expr: true */
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+            sinon.assert.calledOnce(releaseStub);
+            expect(putStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+    });
```

Comment:
Can you also add test case covering https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/274/files#diff-4f6777b6aa31c708f2f14b1e84688d2bR66

Reply:
sure